packages/message-box/__tests__/message-box.spec.ts

```diff
@@ -0,0 +1,183 @@
+      iconClass: 'el-icon-question',
+      message: '这是一段内容',
+    })
+    await sleep(250)
+    const icon = document.querySelector('.el-message-box__status')
+    expect(icon.classList.contains('el-icon-question')).toBe(true)
+  })
+
+  test('html string', async () => {
+    MessageBox({
+      title: 'html string',
+      dangerouslyUseHTMLString: true,
+      message: '<strong>html string</strong>',
+    })
+    await sleep(250)
+    const message = document.querySelector('.el-message-box__message strong')
+    expect(message.textContent).toEqual('html string')
+  })
+
+  test('distinguish cancel and close', async () => {
+    let msgAction = ''
+    MessageBox({
+      title: '消息',
+      message: '这是一段内容',
+      distinguishCancelAndClose: true,
+      callback:  action => {
+        msgAction = action
+      },
+    })
+    await sleep(250)
+    const btn: HTMLElement = document.querySelector('.el-message-box__close')
+    btn.click()
+    await sleep(250)
+    expect(msgAction).toEqual('close')
+  })
+
+  test('alert', async () => {
+    MessageBox.alert('这是一段内容', {
+      title: '标题名称',
+      type: 'warning',
+    })
+    await sleep(250)
+    const vModal: HTMLElement = document.querySelector('.v-modal')
+    vModal.click()
+    await sleep(250)
+    const msgbox: HTMLElement = document.querySelector(selector)
+    expect(msgbox.style.display).toEqual('')
+    expect(msgbox.querySelector('.el-icon-warning')).toBeDefined()
+  })
+
+  test('confirm',  async () => {
+    MessageBox.confirm('这是一段内容', {
+      title: '标题名称',
+      type: 'warning',
+    })
+    await sleep(250)
+    const btn: HTMLElement = document.querySelector(selector).querySelector('.el-button--primary')
+    btn.click()
+    await sleep(250)
+    const msgbox: HTMLElement = document.querySelector(selector)
+    expect(msgbox.style.display).toEqual('none')
+  })
+
+  test('prompt', async () => {
+    MessageBox.prompt('这是一段内容', {
+      title: '标题名称',
+      inputPattern: /test/,
+      inputErrorMessage: 'validation failed',
+    })
+    await sleep(1000)
+    const inputElm: HTMLElement = document.querySelector(selector).querySelector('.el-message-box__input')
+    const haveFocus = inputElm.querySelector('input').isSameNode(document.activeElement)
+    expect(inputElm as HTMLElement).toBeDefined()
+    expect(haveFocus).toBe(true)
+  })
+
+  test('prompt: focus on textarea', () => {
+    // TODO dependencies ElInput
+    MessageBox.prompt('这是一段内容', {
+      inputType: 'textarea',
+      title: '标题名称',
+    })
+  })
+
+  test('callback', async () => {
+    let msgAction = ''
+    MessageBox({
+      title: '消息',
+      message: '这是一段内容',
+      callback: action => {
+        msgAction = action
+      },
+    })
+    await sleep(250)
+    const closeBtn: HTMLElement = document.querySelector('.el-message-box__close')
+    closeBtn.click()
+    await sleep(250)
+    expect(msgAction).toEqual('cancel')
+  })
+
+  test('beforeClose', async() => {
+    let msgAction = ''
```

Comment:
I didn't see any test code in this block. Would you please add missing part here?

Reply:
added